xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-ui/src/main/resources/XWiki/DeletedDocuments.xml

```diff
@@ -341,15 +342,68 @@ XWiki.index.trash.documents.confirmDelete = function(event, table, rowIdx) {
       );
   }
 }
+
+/**
+ * Event listener, called when clicking the "delete all" button.
+ *
+ * @param event the link activation event
+ */
+XWiki.index.trash.documents.confirmDeleteAll = function(event) {
+  event.stop();
+  if (this.disabled) {
+    // Do nothing if the button was already clicked and it's waiting for a response from the server.
+    return;
+  } else {
+    new XWiki.widgets.ConfirmedAjaxRequest(
+      /* Ajax request URL */
+      this.href = "$xwiki.getURL($doc.fullName, 'delete', '&amp;confirm=1&amp;emptybin=true')" + "&amp;form_token=$!{services.csrf.getToken()}" + (Prototype.Browser.Opera ? "" : "&amp;ajax=1"),
+      /* Ajax request parameters */
+      {
+        onCreate : function() {
+          // Disable the button, to avoid a cascade of clicks from impatient users
+          this.disabled = true;
+        }.bind(this),
+        onSuccess : function() {
+          // Update the livetable
+          trashDocumentsLivetable.refresh();
+        }.bindAsEventListener(this),
+        onComplete : function() {
+          // In the end: re-enable the button
+          this.disabled = false;
+        }.bind(this)
+      },
+      /* Interaction parameters */
+      {
+         confirmationText: "$services.localization.render('core.recyclebin.completelyDeleteConfirm')",
+         progressMessageText : "$services.localization.render('platform.index.trashDocumentsDeleteInProgress')",
+         successMessageText : "$services.localization.render('platform.index.trashDocumentsDeleteDone')",
+         failureMessageText : "$services.localization.render('platform.index.trashDocumentsDeleteFailed')"
+      }
+      );
+  }
+}
 /*
  * The livetable is hidden when javascript is disabled, so that an alternative &lt;noscript&gt; table is displayed.
  * Re-display it when the document loads and Javascript is available.
  */
 document.observe('xwiki:dom:loaded', function() {
   if ($('documentsTrash')) {
     $('documentsTrash').up('.hidden').removeClassName('hidden');
+    var actionAllContainer = $('documentsTrash').getElementsBySelector('td.xwiki-livetable-display-header-filter:nth-child(6)')[0];
```

Comment:
Are you sure about this css selector? The element should always be the 6th child if we do that.

Reply:
Yeah that bothers me too, but I don't know another way how to select this specific cell of the livetable. It needs to be the last column of the livetable. Any other idea?

Reply:
```suggestion
    var actionAllContainer = $('documentsTrash').getElementsBySelector('td.xwiki-livetable-display-header-filter:last-child')[0];
```

Reply:
Maybe another better way would be to change the livetable to add a special class to the Action column. I don't know if it can easily be done though.

Reply:
Don't know either. I applied your suggestion for now.